File: test/scala/WordCountTest.scala

import com.tw.samples.ScalaWordCount
import org.apache.spark.{SparkConf, SparkContext}
import org.junit.{After, Before, Test}

class WordCountTest {
    @transient
    protected var sparkContext: SparkContext = null

    @Before
    def setUp() {
        val sparkConf: SparkConf = new SparkConf().setAppName(getClass.getName).setMaster("local")
        sparkContext = new SparkContext(sparkConf)
    }

    @After
    def tearDown() {
        sparkContext.stop()
    }

    @Test def shouldBeAbleToDoWordCountOnTheSimpleStrings() {
        val initialDataset = sparkContext.parallelize(Array("This is a text", "Another text", "More text", "a text"))
        val wordCount = ScalaWordCount.wordCount(initialDataset)
        val result: Array[(String, Int)] = wordCount.collect
        val expected: Array[(String, Int)] = Array(("Another", 1), ("is", 1), ("a", 2), ("text", 4), ("This", 1), ("More", 1))
        assert(result sameElements expected)
    }
}

File: test/java/com/tw/samples/WordCountTest.java
package com.tw.samples;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;
import scala.Tuple2;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class WordCountTest implements Serializable {

    protected transient JavaSparkContext javaSparkContext;

    @Before
    public void setUp() throws Exception {
        SparkConf sparkConf = new SparkConf().setAppName(getClass().getName()).setMaster("local");
        javaSparkContext = new JavaSparkContext(sparkConf);
    }

    @After
    public void tearDown() throws Exception {
        javaSparkContext.close();
    }

    @Test
    public void shouldBeAbleToDoWordCountWithSimpleStrings() {
        JavaRDD<String> initialDataset = javaSparkContext.parallelize(Arrays.asList("This is a text", "Another text", "More text", "a text"));

        JavaPairRDD<String, Integer> wordCount = JavaWordCount.wordCount(initialDataset);

        List<Tuple2<String, Integer>> result = wordCount.collect();

        List<Tuple2<String, Integer>> expected = new ArrayList<Tuple2<String, Integer>>();
        expected.add(new Tuple2<String, Integer>("Another", 1));
        expected.add(new Tuple2<String, Integer>("is", 1));
        expected.add(new Tuple2<String, Integer>("a", 2));
        expected.add(new Tuple2<String, Integer>("text", 4));
        expected.add(new Tuple2<String, Integer>("This", 1));
        expected.add(new Tuple2<String, Integer>("More", 1));

        Assert.assertEquals(expected, result);
    }
}
